File: 2022/02/08/rust/src/climb_stairs.rs
// https://leetcode.com/problems/climbing-stairs/
// Author: Andrew Jarombek
// Date: 2/8/2022

pub fn climb_stairs(n: i32) -> i32 {
    if n < 4 {
        return n;
    }

    let mut prev = 2;
    let mut current = 3;

    for _ in 4..n+1 {
        let temp = current;
        current = prev + current;
        prev = temp;
    }

    return current;
}

File: 2022/03/08/climbStairs.swift
// https://leetcode.com/problems/climbing-stairs/
// Author: Andrew Jarombek
// Date: 3/8/2022

func climbStairs(_ n: Int) -> Int {
    var prev = 1
    var current = 1
    var step = 2

    while step <= n {
        var next = prev + current
        prev = current
        current = next
        step += 1
    }

    return current
}